PR Template: Add a (non-enforcing) checklist of items to consider for… (#10335)
kubevirt/kubevirt: .github/PULL_REQUEST_TEMPLATE.md

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to this PR:

- [ ] Design: A design document was considered and is present (link) or not required

> - [ ] Test: Unit and e2e tests were balanced each other and the right approach chosen

Reviewer: We should add a link to the [reviewer guide "good enough section"](https://github.com/kubevirt/kubevirt/blob/421a6a3acb79204f8fab1e46cb061a33a8f6dbd7/docs/reviewer-guide.md#when-is-a-pr-good-enough) here.

Or we just copy the part into this one.

```suggestion
- [ ] Testing:
  - Unit and e2e tests struck a balance between each other, leaning towards more unit tests and the right approach was chosen
  - Hint: new code requires new unit tests.
  - Hint: new features and bug fixes require at least one e2e test (the core case must be tested).
```

BTW could you elaborate in what you mean with "the right approach was chosen"

Author: > BTW could you elaborate on what you mean with "the right approach was chosen"?

I just tried to appeal at the reader to really think about their decision by including the word *right*. I wanted to them to think twice about what test type to use. But your suggestion is nice, and I'll updat ethe pr - it's much more transparent about what is expected.

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] Maintenance: The [Boy Scout Rule](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) was considered and applied if needed

> - [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test

Reviewer 2: Not every bugfix requires an e2e test, actually most don't, we prefer unit tests :)

Reviewer: A failing e2e test that proves the bug guards against regressions. However, I agree on that we prefer unit tests, so I'd say let's write many unit tests but few e2e tests.

- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required
- [ ] Community: Announce to the [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

> 

Reviewer 2: ```suggestion
- [ ] Upgrades: Does this change affect upgrades?
```

Or something that refers to changes potentially breaking upgrades.

Author: +1

Reviewer: Good point. IIUC then that PR should also be marked with the label `release-note-action-required` and the release note should contain `action required` keywords.

See https://prow.ci.kubevirt.io/command-help#release_note_edit
> PRs that contain 'action required' in their 'release-note' block are given the label 'release-note-action-required' to indicate that the PR introduces potentially breaking changes that necessitate user action before upgrading to the release.

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required

> - [ ] PR: The description will help future contributors

Reviewer 3: ```suggestion
- [ ] PR description: The description will help future contributors
```

Author: I kept PR:, but enhanced the text, thanks!

> - [ ] PR: The description will help fufutre contributors

Reviewer: ```suggestion
- [ ] PR: The description will help future contributors
```

Author: Meh. Thanks!

> - [ ] Maintenance: The [Boy Scout Rule](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) was considered and applied if needed

Reviewer 3: ```suggestion
- [ ] Maintenance: The [Boy Scout Rule](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) was considered and applied if needed. Code should be as clear, easily understandable and simple as possible.
```

Author: I referred to KISS and Fowler which is similar

- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features require at least one e2e test. Bug fixes: It depends
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.

> - [ ] Community: Announce to the [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

Reviewer: ```suggestion
- [ ] Community: Announcement to the [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered
```

Author: Thanks, fixed

Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)

> - [ ] Refactor: The [Boy Scout Rule](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) was considered and applied if needed

Reviewer 3: Awesome! Looks much better now IMO
Thanks!
